samples/nrf5340/netboot/src/main.c

```diff
@@ -39,8 +39,22 @@ int main(void)
 
 	uint32_t s0_addr = s0_address_read();
 	bool valid = false;
+	uint8_t status = pcd_fw_copy_status_get();
 
-	if (pcd_fw_copy_status_get() == PCD_STATUS_COPY) {
+	if (status == PCD_STATUS_READ_VERSION) {
+		err = pcd_find_fw_version();
+		if (err < 0) {
+			printk("Unable to find valid firmware version %d\n\r", err);
+			pcd_fw_copy_invalidate();
+		}
+
+		/* Success, waiting to be rebooted */
+		while (1)
+			;
+		CODE_UNREACHABLE;
```

Comment:
If the version is checked, why would you then want to have to reboot the processor to load a firmware update, surely you'd want to upload it right away after, then reboot?

Reply:
Fair enough. However, the PCD CMD implementation is a bit weird, and I wanted to keep that behaviour the same in this patch. I'm planning and working on reworking the protocol from the ground up. This is a stop-gap to help out projects that needs this functionality before it is ready.